Support entt 3.3.0. Contains incompatible changes and a modified API.

GnikDroy opened the pull request:
I didn't find a dev branch so hopefully we can work on this pull request before merging to master. I have written the demo code as well. I will send a PR on that repository as well.

Please feel free to rename the class/function names since I had it customized for my use.

GnikDroy committed:
Modified readme example code to reflect new API

GnikDroy committed:
Support entt 3.3.0. Contains incompatible changes and a modified API.

Green-Sky commented:
Nice. That was fast.
Now the usual stuff:
- please add yourself to the copyright
- there is a newline missing at the end of the file
- ah yes, the tabs vs spaces war. I'd like them to be tabs :smirk: , sorry

I just noticed the drawback of the new "widget registration", which is that you need to put it in the same namespace (MM). But i guess that's acceptable.

Green-Sky commented on imgui_entt_entity_editor.hpp:
```diff
@@ -1,200 +1,156 @@
+        auto index = entt::type_info<Component>::id();
+        auto [it, insert_result] = component_infos.insert_or_assign(index, component_info);
+        assert(insert_result);
+        auto& [type_index, info] = *it;
```
we have an unused variable, how about something like:
```cpp
return std::get<ComponentInfo>(*it);
```

Green-Sky commented on imgui_entt_entity_editor.hpp:
```diff
@@ -1,200 +1,156 @@
+                ImGui::Separator();
+
+                if (registry.valid(e)) {
+                    const std::string entity_id = std::to_string(static_cast<int>(registry.entity(e))) + std::to_string(static_cast<int>(registry.version(e)));
```
I think `entt::to_integral(e)` would suffice.

Green-Sky replied:
used to be `entt::to_integer()`

GnikDroy replied:
Yes. That would indeed be better than the current method.

GnikDroy committed:
Removed unused variable.
Using entt::integral instead of static_cast<int>.
Using tabs for indentation instead of spaces.

GnikDroy committed:
Add newline at the end.

GnikDroy commented:
I have made all the requested changes. : )

GnikDroy committed:
Using ImGui::PushID() and ImGui::PopID() instead of manually creating scopes.
It is no longer required to add a scope to every ImGui widget in the custom component widgets.
Modified README to reflect the changes in the way to write custom widgets.

Green-Sky commented:
nice idea

Green-Sky reviewed (approved):
so, i had a test run, and it looks like it runs.
